### PR TITLE
Do not merge: scratch work 2

### PR DESCRIPTION
**Pull request by charlesccychen**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: sdks/python/apache_beam/io/gcp/pubsub.py [hunk 1]
```diff
@@ -28,6 +28,9 @@
 from apache_beam.io.iobase import Read
 from apache_beam.io.iobase import Write
 from apache_beam.runners.dataflow.native_io import iobase as dataflow_io
+from apache_beam.transforms import core
+from apache_beam.transforms import window
+from apache_beam.transforms import Map
```
Comment (charlesccychen) [inline sdks/python/apache_beam/io/gcp/pubsub.py]: ok

File: sdks/python/apache_beam/pipeline.py [hunk 2]
```diff
@@ -161,7 +161,7 @@ def run(self, test_runner_api=True):
     """Runs the pipeline. Returns whatever our runner returns after running."""
 
     # When possible, invoke a round trip through the runner API.
```
Comment (cosmoskitten) [inline sdks/python/apache_beam/pipeline.py]: Please fix

Comment (charlesccychen) [inline sdks/python/apache_beam/pipeline.py]: > **cosmoskitten** wrote:
> Please fix


Done.

File: sdks/python/apache_beam/runners/direct/bundle_factory.py [hunk 3]
```diff
@@ -38,7 +38,10 @@ def __init__(self, stacked):
     self._stacked = stacked
 
   def create_bundle(self, output_pcollection):
-    return _Bundle(output_pcollection, self._stacked)
+    return _Bundle(output_pcollection, None, stacked=self._stacked)
+
+  def create_keyed_bundle(self, output_pcollection, key):
```
Comment (charlesccychen) [inline sdks/python/apache_beam/runners/direct/bundle_factory.py]: Please fix create_keyed_bundle.

Comment (aaltay) [inline sdks/python/apache_beam/runners/direct/bundle_factory.py]: > **charlesccychen** wrote:
> Please fix create_keyed_bundle.


Done.

**Pull request closed without merging**